File: Backend/FPAMS/FPAMS.Domain/Entities/Designation.cs
using FPAMS.Domain.Common;

namespace FPAMS.Domain.Entities;

public class Designation : BaseEntity
{
    public string DesignationCode { get; set; } = string.Empty;

    public string DesignationName { get; set; } = string.Empty;

    public string? Description { get; set; }

    public int DisplayOrder { get; set; }

    public bool IsActive { get; set; } = true;

    public ICollection<User> Users { get; set; } = new List<User>();
}

File: Frontend/fpams-web/src/types/designation.ts
export interface Designation {
    id: string;
    designationCode: string;
    designationName: string;
    description?: string | null;
    displayOrder: number;
    isActive: boolean;
}

export interface CreateDesignationRequest {
    designationCode: string;
    designationName: string;
    description?: string | null;
    displayOrder: number;
}

export interface UpdateDesignationRequest {
    id: string;
    designationCode: string;
    designationName: string;
    description?: string | null;
    displayOrder: number;
    isActive: boolean;
}
